wpilibj/src/main/java/edu/wpi/first/wpilibj/util/Color.java

```diff
@@ -185,6 +156,190 @@ private static double roundAndClamp(double value) {
+    // is one rgb component at a low value (m), one at a high value (v) and one
+    // that changes (X) from low to high (X+m) or high to low (v-X)
+
+    // Difference between highest and lowest value of any rgb component
+    final int chroma = (s * v) / 255;
+
+    // Because hue is 0-180 rather than 0-360 use 30 not 60
+    final int region = (h / 30) % 6;
+
+    // Remainder converted from 0-30 to 0-255
+    final int remainder = (int) Math.round((h % 30) * (255 / 30.0));
+
+    // Value of the lowest rgb component
+    final int m = v - chroma;
+
+    // Goes from 0 to chroma as hue increases
+    final int X = (chroma * remainder) >> 8;
+
+    int red;
+    int green;
+    int blue;
+    switch (region) {
+      case 0:
+        red = v;
+        green = X + m;
+        blue = m;
+        break;
+      case 1:
+        red = v - X;
+        green = v;
+        blue = m;
+        break;
+      case 2:
+        red = m;
+        green = v;
+        blue = X + m;
+        break;
+      case 3:
+        red = m;
+        green = v - X;
+        blue = v;
+        break;
+      case 4:
+        red = X + m;
+        green = m;
+        blue = v;
+        break;
+      default:
+        red = v;
+        green = m;
+        blue = v - X;
+        break;
+    }
+    return packRGB(red, green, blue);
+  }
+
+  /** Represents a color channel in an RGB color. */
+  public enum RGBChannel {
+    /** The red channel of an RGB color. */
+    kRed,
+    /** The green channel of an RGB color. */
+    kGreen,
+    /** The blue channel of an RGB color. */
+    kBlue
+  }
+
+  /**
+   * Packs 3 RGB values into a single 32-bit integer. These values can be unpacked with
+   * {@link #unpackRGB(int, RGBChannel)} to retrieve the values. This is helpful for avoiding memory
+   * allocations of new {@code Color} objects and its resulting garbage collector pressure.
+   *
+   * @param r the value of the first channel to pack
+   * @param g the value of the second channel to pack
+   * @param b the value of the third channel to pack
+   * @return the packed integer
+   */
+  public static int packRGB(int r, int g, int b) {
+    return (r & 0xFF) << 16 | (g & 0xFF) << 8 | (b & 0xFF);
+  }
+
+  /**
+   * Unpacks a single color channel from a packed 32-bit RGB integer.
+   *
+   * <p>Note: Packed RGB colors are expected to be in byte order [empty][red][green][blue].
+   *
+   * @param packedColor the packed color to extract from
+   * @param channel the color channel to unpack
+   * @return the value of the stored color channel
+   */
+  public static int unpackRGB(int packedColor, RGBChannel channel) {
+    switch (channel) {
+      case kRed:
+        return (packedColor >> 16) & 0xFF;
+      case kGreen:
+        return (packedColor >> 8) & 0xFF;
+      case kBlue:
+        return packedColor & 0xFF;
+      default:
+        DriverStation.reportWarning("Unknown color channel for unpacking: " + channel, true);
```

Comment:
Raising an error would work here, too; the default case will only be hit if a new enum entry is added without a corresponding case in the switch statement. However, Java 14 (released in 2020!) added switch expressions that will result in compile-time errors if not every branch's is covered. Assuming we upgrade to a modern version of Java for 2025, I'm not particularly concerned about the default case behavior because it'll just end up getting removed